editoast/src/schema/buffer_stop.rs

```diff
@@ -23,6 +30,74 @@ pub struct BufferStop {
     pub applicable_directions: ApplicableDirections,
 }
 
+#[derive(QueryableByName, Queryable, Debug, Clone)]
+pub struct BufferStopQueryable {
+    #[sql_type = "Integer"]
+    pub id: i32,
+    #[sql_type = "Text"]
+    pub obj_id: String,
+    #[sql_type = "Json"]
+    pub data: Value,
+    #[sql_type = "Integer"]
+    pub infra_id: i32,
+}
+
+impl From<BufferStopQueryable> for BufferStop {
+    fn from(buffer_stop_queryable: BufferStopQueryable) -> Self {
+        let buffer_stop: BufferStop = serde_json::from_value(buffer_stop_queryable.data).unwrap();
+        Self {
+            id: buffer_stop.id,
+            track: buffer_stop.track,
+            position: buffer_stop.position,
+            applicable_directions: buffer_stop.applicable_directions,
+        }
+    }
+}
+
+impl BufferStop {
+    pub fn persist(
+        values: &[Self],
+        infrastructure_id: i32,
+        conn: &PgConnection,
+    ) -> Result<(), Box<dyn ApiError>> {
+        use crate::tables::osrd_infra_bufferstopmodel::dsl::*;
+
+        let datas = values
+            .iter()
+            .map(|value| {
+                (
+                    obj_id.eq(value.get_id().clone()),
+                    data.eq(serde_json::to_value(value).unwrap()),
+                    infra_id.eq(infrastructure_id),
+                )
+            })
+            .collect::<Vec<_>>();
+
+        diesel::insert_into(osrd_infra_bufferstopmodel)
+            .values(datas)
+            .execute(conn)?;
+
+        Ok(())
+    }
+
+    pub fn find_by_infrastructure_id(
+        conn: &PgConnection,
+        infrastructure_id: &i32,
+    ) -> Result<Vec<Self>, Box<dyn ApiError>> {
+        use crate::tables::osrd_infra_bufferstopmodel::dsl::*;
+
+        let buffer_stops: Vec<BufferStop> = osrd_infra_bufferstopmodel
+            .filter(infra_id.eq(infrastructure_id))
+            .load::<BufferStopQueryable>(conn)
+            .expect("find_by_infrastructure_id failed")
+            .into_iter()
+            .map(|e| e.into())
+            .collect();
+
+        Ok(buffer_stops)
+    }
+}
```

Comment:
Not used

Reply:
well seen ! was added in the wrong struct at first

Reply:
not now but will be used in some further asserts that are missing.

Reply:
done